nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/request/TokenRequest.java

```diff
@@ -49,6 +49,24 @@ public TokenRequest(HttpServletRequest request) {
         clientSecret = request.getParameter(CLIENT_SECRET_PARAM);
         refreshToken = request.getParameter(REFRESH_TOKEN_PARAM);
         codeVerifier = request.getParameter(CODE_VERIFIER_PARAM);
+
+        checkAuthorization(request);
+    }
+
+    protected void checkAuthorization(HttpServletRequest request) {
+        final String authorization = request.getHeader("Authorization");
+        if (authorization != null && authorization.toLowerCase().startsWith("basic")) {
+            // Authorization: Basic base64credentials
+            String base64Credentials = authorization.substring("Basic".length()).trim();
```

Comment:
Could you use `"basic "` (with space) in both cases? And put that in a constant?